src/serialize.rs

```diff
@@ -57,61 +70,99 @@ impl PrometheusSerializer {
+    }
+
+    #[cfg(feature = "resource_selector")]
+    fn generate_static_labels(
+        &self,
+        resource: &Resource,
+    ) -> std::io::Result<Option<SmartString<smartstring::LazyCompact>>> {
+        let mut has_attrs = false;
+        let mut buf = SmartString::<smartstring::LazyCompact>::new();
+        if !matches!(self.config.resource_selector, ResourceSelector::None) {
+            for (key, value) in resource
+                .iter()
+                .filter(|attr| self.config.resource_selector.matches(attr.0))
+            {
+                let sanitized_key = sanitize_name(key.as_str());
+                // We need a string representation beforehand, as we rely on debug formatter
+                // to properly escape special or non-printable characters.
+                let string_value = value.as_str();
+                if !has_attrs {
+                    has_attrs = true;
+                    write!(&mut buf, "{sanitized_key}={string_value:?}")
+                } else {
+                    write!(&mut buf, ",{sanitized_key}={string_value:?}")
+                }
+                .map_err(std::io::Error::other)?;
+            }
+        }
+        if has_attrs { Ok(Some(buf)) } else { Ok(None) }
     }
 
     fn serialize_resource_metrics<W: Write>(
         &self,
         rm: &ResourceMetrics,
-        writer: &mut W,
+        context: &mut SerializerContext<W>,
     ) -> std::io::Result<()> {
         // Serialize all scope metrics first
         for sm in rm.scope_metrics() {
-            self.serialize_scope_metrics(sm, writer)?;
+            self.serialize_scope_metrics(sm, context)?;
         }
 
         // Serialize resource as target_info
-        self.serialize_resource(rm.resource(), writer)?;
+        self.serialize_resource(rm.resource(), context)?;
 
         Ok(())
     }
 
     fn serialize_resource<W: Write>(
         &self,
         resource: &Resource,
-        writer: &mut W,
+        context: &mut SerializerContext<W>,
     ) -> std::io::Result<()> {
         // Don't serialize empty resources or if target_info is disabled
         if resource.is_empty() || self.config.disable_target_info {
             return Ok(());
         }
 
-        write_type_comment(writer, "target_info", "gauge")?;
-        write_help_comment(writer, "target_info", "Target metadata")?;
+        write_type_comment(context.writer, "target_info", "gauge")?;
+        write_help_comment(context.writer, "target_info", "Target metadata")?;
 
-        write!(writer, "target_info")?;
+        write!(context.writer, "target_info")?;
 
-        let mut label_writer = LabelWriter::new(writer);
+        let mut label_writer = LabelWriter::new(context);
         for (key, value) in resource.iter() {
+            // Skip attributes that were added to `static_labels`
+            #[cfg(feature = "resource_selector")]
+            if self.config.resource_selector.matches(key) {
+                continue;
+            }
```

Comment:
That wasn't the case in `opentelemetry-prometheus`, not sure if we want this to be the case? I don't really see why you would want those resources to be skipped in the `target_info`?